lib/customers.js

```diff
@@ -115,17 +117,32 @@ function getById (id, userToken) {
  */
 function getDailyVolume (id) {
   return Promise.all([
-    db.one(`select coalesce(sum(fiat), 0) as total from cash_in_txs
+    db.one(`select coalesce(sum(fiat), 0) as total, max(created) as maxdate from cash_in_txs
            where customer_id=$1
            and created > now() - interval '1 day'`, [id]),
-    db.one(`select coalesce(sum(fiat), 0) as total from cash_out_txs
+    db.one(`select coalesce(sum(fiat), 0) as total, max(created) as maxdate from cash_out_txs
            where customer_id=$1
            and created > now() - interval '1 day'`, [id])
-  ]).then(([cashInTotal, cashOutTotal]) => {
-    return BN(cashInTotal.total).add(cashOutTotal.total)
+  ]).then(([cashIn, cashOut]) => {
+    const dailyVolume = BN(cashIn.total).add(cashOut.total)
+    const hoursTillLimitClear = getHoursTillLimitClear(cashIn.maxdate, cashOut.maxdate)
+    return { dailyVolume, hoursTillLimitClear }
   })
 }
 
+function getHoursTillLimitClear (cashInDate, cashOutDate) {
+  let startDate = moment()
+  startDate = startDate.subtract(1, 'days')
+
+  const cashInMoment = moment(cashInDate || startDate)
+  const cashOutMoment = moment(cashOutDate || startDate)
+
+  const cashInDuration = moment.duration(cashInMoment.diff(startDate)).asHours()
+  const cashOutDuration = moment.duration(cashOutMoment.diff(startDate)).asHours()
+
+  return _.ceil(_.max([cashInDuration, cashOutDuration]))
```

Comment:
Do we need to check for negative values?

Reply:
Not really because of the sql 'where'.

Reply:
Hmm, maybe but I think an extra check in the logic here will make it safer.

Reply:
You're absolutely right =)